velox/docs/develop/aggregate-functions.rst

```diff
@@ -169,6 +187,15 @@ function's argument type(s) wrapped in a Row<> even if the function only takes
 one argument. This is needed for the SimpleAggregateAdapter to parse input
 types for arbitrary aggregation functions properly.
 
+A FunctionState struct needs to be declared in the simple aggregation function
+class, it is used to hold the function-level variables that are typically
+computed once and used at every row when adding inputs to accumulators or
+extracting values from accumulators. For example, if the UDAF needs to get the
+result type or the raw input type of the aggregaiton function, the author can
+hold them in the FunctionState struct, and initialize them in the initialize()
+method. If the UDAF does not require the use ofFunctionState, it is necessary
```

Comment:
nit: Let's use aggregation function instead of UDAF in this documentation to be consistent with the existing content.

ofFunctionState --> of FunctionState